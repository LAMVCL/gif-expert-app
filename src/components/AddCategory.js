import React, { useState } from 'react';
import PropTypes from 'prop-types';

//Desestructuramos el prop mediante llaves {}
export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);

    }
    const handleSubmit = (e) =>{
        //Bloqueamos la actualización de la página
        e.preventDefault();

        //Validamos que el input posea texto mediante un length
        if (inputValue.trim().length > 2){
            //De ser valido asignamos el texto al array mediante setCategories
            setCategories(newCategorie => [inputValue,...newCategorie ]);
            //Vaciamos el input para evitar duplicados al presionar doble submit
            setInputValue('');
        }
        
    }
    return (
        <form className='formulario' onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
            type="text"
            value={inputValue}
            placeholder='Buscar...'
            onChange={handleInputChange}
            />
            <button ><i className="fas fa-search"></i></button>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
