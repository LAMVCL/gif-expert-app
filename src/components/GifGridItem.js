import React from 'react'

export const GifGridItem = ({id, title, url}) => {

    return (
        <div className="card animate__animated animate__fadeInDown">
            <p>{title}</p>
            <img src={url} alt={title}></img>
        </div>
    )
}
