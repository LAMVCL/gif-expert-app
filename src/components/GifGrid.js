import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() =>{

        const urlFetch = `https://api.giphy.com/v1/gifs/search?q=${category}&limit=10&api_key=g1dCiL4mecdpaOWgTGtlvZzrLr0c9nBW`;
        const respuesta = await fetch(urlFetch);
        //Destructuramos data para evitar poner data.data
        const {data} = await respuesta.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized.url,
            }
        });

        console.log(gifs);
        //Le pasamos el array de 10 imagenes con su id, titulo y url
        setImages(gifs);
    }

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                
                {
                    images.map( (img) => {

                        return <GifGridItem key={img.id} {...img}
                        />
                    })
                }
            </div>
        </>
        
    )
}
