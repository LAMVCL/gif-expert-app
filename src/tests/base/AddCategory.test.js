import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";


describe('Set de pruebas de AddCategory', () => {

    const setCategories = jest.fn();
    //se inicializa para el intellisense
    let wrapper;
    
    beforeEach(() =>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });
    
    test('prueba de snapshot en <AddCategory/> ', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de cambiar la caja de texto ', () => {
        
        const input = wrapper.find('input');

        //Le mandamos  el evento
        const value = 'Valor de evento';
        input.simulate('change',{
            target: {
                value: value
            }
        });

        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('NO debe de postear la informaciÃ³n con submit', () => {

        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });
    
    
    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {

        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
        //1.-simular el imputChange
        //Obtenemos la caja de texto.
        const input = wrapper.find('input');
        //Le mandamos  el evento
        const value = 'Hola mundo';
        input.simulate('change',{
            target: {
                value: value
            }
        });

        //2.- Simular el submit
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        
        //3.- setCategories se debe haber llamado
        expect(setCategories).toHaveBeenCalled();
        //4.- Debe pasar una funcion
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        //5.- el valor del input debe estar ''
        expect(wrapper.find('input').prop('value')).toBe('');
    });
    
});
