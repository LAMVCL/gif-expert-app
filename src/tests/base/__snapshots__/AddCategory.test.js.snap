// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set de pruebas de AddCategory prueba de snapshot en <AddCategory/>  1`] = `
<form
  className="formulario"
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Buscar..."
    type="text"
    value=""
  />
  <button>
    <i
      className="fas fa-search"
    />
  </button>
</form>
`;
